config/IBAMR/packages/SAMRAI.py

```diff
@@ -6,11 +6,18 @@ def __init__(self, framework):
     self.download  = ['http://ftp.mcs.anl.gov/pub/petsc/externalpackages/SAMRAI-v2.4.4p.tar.gz', 'https://computation-rnd.llnl.gov/SAMRAI/download/SAMRAI-v2.4.4.tar.gz']
     self.functions = []
     self.includes  = ['Box.h']
-    self.liblist   = [['libSAMRAI.a', 'libSAMRAI2d_algs.a', 'libSAMRAI2d_appu.a', 'libSAMRAI2d_geom.a', 'libSAMRAI2d_hier.a', 'libSAMRAI2d_math_std.a', 'libSAMRAI2d_mesh.a', 'libSAMRAI2d_pdat_std.a', 'libSAMRAI2d_solv.a', 'libSAMRAI2d_xfer.a']]
+    if framework.argDB['DIM'] == 2:
```

Comment:
I do not think querying the argDB in __init__ works. Configure creates all the nodes, then it calls setupHelp() on all of them, then it calls setup(), and then it calls configure(). I think you need to check this in setup() or configure(). I would lean towards the latter.